cluster_library.c

```diff
@@ -1011,17 +984,16 @@ void cluster_init_cache(redisCluster *c, redisCachedCluster *cc) {
     for (i = 0; i < cc->count; i++) map[i] = i;
     fyshuffle(map, cc->count);
 
+    /* Duplicate the hash key so we can invalidate when redirected */
+    c->cache_key = zend_string_dup(cc->hash, 0);
```

Comment:
Why do you use `zend_string_dup` instead of `zend_string_copy`?

Reply:
I was using `zend_string_copy` but it was causing havoc when invalidating the cache.

I'll test again with the other changes in place.  I don't think it's a big deal to duplicate the string since these seeds are small.

Reply:
OK switched back to `zend_string_copy` and everything seems good :+1: